### ISSUE TITLE
(MATLAB) Unrecognized function or variable 'stirMEX'.

### ISSUE THREAD

**Issue by mirtiny** [UCL/STIR#1210]
I followed the instructions to the best of my ability on this page here:
https://stir.sourceforge.net/wiki/index.php/Installing_STIR_with_CMake#Visual_Studio.2FXCode_and_other_systems_with_an_IDE_for_compilation

I believe I've done everything correctly, however, when trying to use STIR with Matlab I'm met with an error (title). Did I miss a step or something? Here's what my directory looks like for Matlab.

![image](https://github.com/UCL/STIR/assets/131629965/7b342f3b-738f-4096-b352-bbec932be542)

**Comment by KrisThielemans** [comment]
a long time ago I've tried to build with MATLAB, but is this directory in your MATLAB path?

**Comment by mirtiny** [comment]
Yeah I think so. I ran this command in my Matlab console:
 `addpath C:/STIR/matlab`

And just to confirm that it's in my path, running some of the functions in that directory (i.e `SwigGet`) will fully run fine.

**Comment by mirtiny** [comment]
Also, this may be possibly related but I had to edit the stirMATLAB_wrap.cxx file to fix the following errors:
![image](https://github.com/UCL/STIR/assets/131629965/9d7b3760-f0e4-4c80-ab9b-cbcfbc28d75d)

Before:
![image](https://github.com/UCL/STIR/assets/131629965/53c9fbf3-4fc9-4589-baf3-43840629051b)

After (Concatenated the two strings manually):
![image](https://github.com/UCL/STIR/assets/131629965/3bd0abcf-73d7-4f4c-bf50-539b19888ecf)

**Comment by KrisThielemans** [comment]
> I ran this command in my Matlab console:
> `addpath C:/STIR/matlab`

great. Can you then give an example + error message of what you're actually doing/seeing? Did you see the examples in https://github.com/UCL/STIR/tree/master/examples/matlab?

> this may be possibly related but I had to edit the stirMATLAB_wrap.cxx file to fix the following errors:

Interesting. This will be because the SWIG matlab branch is very out-of-date. However, if everything compiled fine after your fix, I don't think this is the problem.

**Comment by mirtiny** [comment]
> great. Can you then give an example + error message of what you're actually doing/seeing?
I typed in `stir` in the command window. After some testing, I realized that this is possibly intended and that I need to run a stir function (ie `stir.OSMAPOSLReconstruction3DFloat('xxxxx')`).

> Did you see the examples in https://github.com/UCL/STIR/tree/master/examples/matlab?
I tried a few of them, I'm getting the same error anytime a stir function is used.

(recon_demo.m)
![image](https://github.com/UCL/STIR/assets/131629965/45a64144-c74a-42cb-b75b-45ae154b0558)
